src/change_stream/options.rs

```diff
@@ -13,95 +12,28 @@ use crate::{
     selection_criteria::SelectionCriteria,
 };
 
-/// These are the valid options that can be passed to the `watch` method for creating a
-/// [`ChangeStream`](crate::change_stream::ChangeStream).
 #[skip_serializing_none]
-#[derive(Clone, Debug, Default, Deserialize, Serialize, TypedBuilder)]
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
 #[non_exhaustive]
-pub struct ChangeStreamOptions {
-    #[rustfmt::skip]
-    /// Configures how the
-    /// [`ChangeStreamEvent::full_document`](crate::change_stream::event::ChangeStreamEvent::full_document)
-    /// field will be populated. By default, the field will be empty for updates.
-    #[builder(default)]
-    pub full_document: Option<FullDocumentType>,
-
-    /// Configures how the
-    /// [`ChangeStreamEvent::full_document_before_change`](
-    /// crate::change_stream::event::ChangeStreamEvent::full_document_before_change) field will be
-    /// populated.  By default, the field will be empty for updates.
-    #[builder(default)]
-    pub full_document_before_change: Option<FullDocumentBeforeChangeType>,
-
-    /// Specifies the logical starting point for the new change stream. Note that if a watched
-    /// collection is dropped and recreated or newly renamed, `start_after` should be set instead.
-    /// `resume_after` and `start_after` cannot be set simultaneously.
-    ///
-    /// For more information on resuming a change stream see the documentation [here](https://www.mongodb.com/docs/manual/changeStreams/#change-stream-resume-after)
-    #[builder(default)]
-    pub resume_after: Option<ResumeToken>,
-
-    /// The change stream will only provide changes that occurred at or after the specified
-    /// timestamp. Any command run against the server will return an operation time that can be
-    /// used here.
-    #[builder(default)]
-    pub start_at_operation_time: Option<Timestamp>,
-
-    /// Takes a resume token and starts a new change stream returning the first notification after
-    /// the token. This will allow users to watch collections that have been dropped and
-    /// recreated or newly renamed collections without missing any notifications.
-    ///
-    /// This feature is only available on MongoDB 4.2+.
-    ///
-    /// See the documentation [here](https://www.mongodb.com/docs/master/changeStreams/#change-stream-start-after) for more
-    /// information.
-    #[builder(default)]
-    pub start_after: Option<ResumeToken>,
-
-    /// If `true`, the change stream will monitor all changes for the given cluster.
-    #[builder(default, setter(skip))]
+pub(crate) struct ChangeStreamOptions {
```

Comment:
Unlike `list_databases`, the situation with options and operations is a little more messy here, so I opted to simply make this non-`pub` and defer further internal refactoring.